src/components/Header/Header.tsx

```diff
@@ -57,7 +57,7 @@ const MemoizedHeader = memo(
     email: string;
     isInternal?: boolean;
   }) => {
-    const search = new URLSearchParams(window.location.search).keys().next().value;
+    const search = new URLSearchParams(window.location.search).keys().next().value ?? '';
```

Comment:
I am not sure if this change is needed.

But if it is, I'd not use window.location. Can you use the useSearchParams hook from react router? Using location does not trigger re-renders if necessary.